src/TrackPanel.cpp

```diff
@@ -312,14 +314,24 @@ TrackPanel::TrackPanel(wxWindow * parent, wxWindowID id,
    auto theProject = GetProject();
    theProject->Bind(
       EVT_PROJECT_SETTINGS_CHANGE, &TrackPanel::OnProjectSettingsChange, this);
-   theProject->Bind(
-      EVT_TRACK_FOCUS_CHANGE, &TrackPanel::OnTrackFocusChange, this );
+   mFocusChangeSubscription = TrackFocus::Get(*theProject)
+      .Subscribe(*this, &TrackPanel::OnTrackFocusChange);
 
    mUndoSubscription = UndoManager::Get(*theProject)
       .Subscribe(*this, &TrackPanel::OnUndoReset);
 
    mAudioIOSubscription =
       AudioIO::Get()->Subscribe(*this, &TrackPanel::OnAudioIO);
+
+   mRealtimeEffectManagerSubscription = RealtimeEffectManager::Get(*theProject)
+      .Subscribe([this](const RealtimeEffectManagerMessage& msg)
+      {
+         if(!msg.track)
+            Refresh();
```

Comment:
I understand you need to repaint a track if effects are inserted or removed in its stack, because the button may need to change its appearance when the effect list becomes empty or becomes nonempty.

I do not see why you need to refresh the whole panel when there are changes in the global list of effects.

No harm would be done, but a needless repainting.